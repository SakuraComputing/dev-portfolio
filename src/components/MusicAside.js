import React from 'react';

export const MusicAside = () => (
    <aside className="list music aside section">
        <div className="section-inner">
            <h2 className="heading">Favourite coding music</h2>
            <div className="content">
                <ul className="list-unstyled">
                    <li><i className="fa fa-headphones"></i> <a href="#">Etiam hendrerit urna nunc</a></li>
                    <li><i className="fa fa-headphones"></i> <a href="#">Ut sollicitudin in mauris non auctor</a></li>
                    <li><i className="fa fa-headphones"></i> <a href="#">Etiam hendrerit urna nunc</a></li>
                    <li><i className="fa fa-headphones"></i> <a href="#">Duis et felis bibendum</a></li>
                </ul>
            </div>
        </div>
    </aside>
);

export default MusicAside;