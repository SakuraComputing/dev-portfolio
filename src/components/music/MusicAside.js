import React from 'react';
import Songs from './Songs';

class MusicAside extends React.Component {

    render() {

        const { musicaside } = this.props;

        let musicItems;

        if(musicaside.length > 0 ) {
            musicItems = musicaside.map(song => (
                <Songs key={song.id} song={song}/>
            ))
        } else {
            musicItems = <h4>No Songs Found.......</h4>
        }

        return (
            <aside className="list music aside section">
                <div className="section-inner">
                    <h2 className="heading">Favourite coding music</h2>
                    <div className="content">
                        <ul className="list-unstyled">
                            {musicItems}
                        </ul>
                    </div>
                </div>
            </aside>
        );
    }
}

export default MusicAside;
