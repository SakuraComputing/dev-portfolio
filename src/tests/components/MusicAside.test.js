import React from 'react';
import { shallow } from 'enzyme';
import MusicAside from '../../components/MusicAside';

describe('MusicAside Section', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<MusicAside />);
        expect(wrapper).toMatchSnapshot();
    });
});