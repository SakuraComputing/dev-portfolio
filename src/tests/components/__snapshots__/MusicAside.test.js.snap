// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicAside Section should render correctly 1`] = `
<aside
  className="list music aside section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Favourite coding music
    </h2>
    <div
      className="content"
    >
      <ul
        className="list-unstyled"
      >
        <li>
          <i
            className="fa fa-headphones"
          />
           
          <a
            href="#"
          >
            Etiam hendrerit urna nunc
          </a>
        </li>
        <li>
          <i
            className="fa fa-headphones"
          />
           
          <a
            href="#"
          >
            Ut sollicitudin in mauris non auctor
          </a>
        </li>
        <li>
          <i
            className="fa fa-headphones"
          />
           
          <a
            href="#"
          >
            Etiam hendrerit urna nunc
          </a>
        </li>
        <li>
          <i
            className="fa fa-headphones"
          />
           
          <a
            href="#"
          >
            Duis et felis bibendum
          </a>
        </li>
      </ul>
    </div>
  </div>
</aside>
`;
